### DISCUSSION TITLE
Is there any cost involved while scraping metrics from kubernetes

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#5014] [General]
Hi I have implemented a monitoring setup by following this link https://strimzi.io/docs/operators/latest/deploying.html#assembly-metrics-str . I have a doubt that prometheus server will scrape metrics from kubernetes [kubernetes-cadvisor, kubernetes kubelet]. I am using Azure kubernetes service, so as per my understanding periodic api calls will made to kubernetes to get the metrics. So my question is that is there any cost involved for these api calls or any other cost is involved?
If so where can I find the cost estimation?

And are there any more grafana dashboards which are strimzi specific. Suppose If I want to add more panels to the strimzi-kafka dashboard how can I do it?

**Comment by scholzj** [comment]
Every API call has some cost. But I do not think this is the right forum to understand how big the cost it. Kubernetes needs to provide some additional metrics such as available disk space on the volumes, resource utilisation etc. So if you do not have these working, they might be missing in the dashboards. But it of course does not affect Kafka directly.
